### PR TITLE
Address items in 14279

### PR DESCRIPTION
**Pull request by chrisdecenzo**
#### Problem
* Address Issue 14279

#### Change overview
* removed unneeded includes
* changed CatalogVendorApp function arguments from object to const pointers
* fix recent regression that broke commissioning from tv-app
* remove CreateBindingWithCallback from CHIPDeviceController

#### Testing
* performed tv-app and tv-casting-app commissioning
* assorted tv-app and chip-tool cluster commands

### CODE REVIEW

File: examples/tv-app/linux/AppImpl.cpp [hunk 1]
```diff
@@ -168,10 +168,11 @@ CHIP_ERROR ContentAppFactoryImpl::LookupCatalogVendorApp(uint16_t vendorId, uint
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ContentAppFactoryImpl::ConvertToPlatformCatalogVendorApp(CatalogVendorApp sourceApp, CatalogVendorApp * destinationApp)
+CHIP_ERROR ContentAppFactoryImpl::ConvertToPlatformCatalogVendorApp(const CatalogVendorApp * sourceApp,
```
Comment (bzbarsky-apple) [inline examples/tv-app/linux/AppImpl.cpp:171]: Just to check, why not:
```suggestion
CHIP_ERROR ContentAppFactoryImpl::ConvertToPlatformCatalogVendorApp(const CatalogVendorApp & sourceApp,
```
?

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by andy31415** [comment]
@chrisdecenzo - is this still a draft as the title says? the issue is not marked as draft anymore, but title says it is.

File: examples/tv-app/linux/AppImpl.cpp (hunk 1 above)
Comment (chrisdecenzo) [reply examples/tv-app/linux/AppImpl.cpp]: good feedback, updated to follow this convention

**Comment by chrisdecenzo** [comment]
> @chrisdecenzo - is this still a draft as the title says? the issue is not marked as draft anymore, but title says it is.

This is no longer a draft. Working now to get CI to succeed.

**Review by lazarkov — APPROVED** [review]

File: examples/platform/linux/AppMain.cpp [hunk 2]
```diff
@@ -371,41 +376,69 @@ void PairingCommand::OnPairingDeleted(CHIP_ERROR err)
...
-        EndpointId endpointId = 1;
-        ClusterId clusterId   = kInvalidClusterId;
+void PairingCommand::OnDeviceConnectedFn(void * context, chip::OperationalDeviceProxy * device)
+{
+    ChipLogProgress(Controller, "OnDeviceConnectedFn");
+    CommissionerDiscoveryController * cdc = GetCommissionerDiscoveryController();
 
-        gCommissioner.CreateBindingWithCallback(nodeId, kBindingClusterEndpoint, gLocalId, groupId, endpointId, clusterId,
-                                                OnSuccessResponse, OnFailureResponse);
+    if (device == nullptr)
```
Comment (bzbarsky-apple) [inline examples/platform/linux/AppMain.cpp:411]: I really don't think that can happen...  But either way.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by xylophone21 — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**